strapi/strapi
# Enhancement: Improve support and tests for long identifiers in the database (without enabling)

### What does it do?

- removes the review workflow workaround that was previously used
- fixes issue where base table name for content types was being snake_case where it previously wasn't in Strapi 4
- enforces an explicit maxLength passed to all the naming functions
- improves testing of createMetadata and naming functions

Still remaining TODO: write a migration and turn the feature on by default

### Why is it needed?

Tracking issue: https://github.com/strapi/strapi/issues/13117

### How to test it?

Test that it works:
- Starting with this PR, enable the feature by changing MAX_DB_IDENTIFIER_LENGTH to 55
- Everything in Strapi should work fine
- Create content types, attributes, and every type of relation with names of 60 characters
- There should be no index, column, or table in the database with a name longer than 55 characters

Test that it still hasn't changed anything (feature is not on):
- create a v4 project
- switch to this PR (do not update the identifier length constant)
- everything should work exactly as before with no data loss

### Related issue(s)/PR(s)

Tracking issue: https://github.com/strapi/strapi/issues/13117

Previous PRs to lay the foundation:
https://github.com/strapi/strapi/pull/19608
https://github.com/strapi/strapi/pull/19596
https://github.com/strapi/strapi/pull/19567
https://github.com/strapi/strapi/pull/19561
https://github.com/strapi/strapi/pull/19560
https://github.com/strapi/strapi/pull/19460
